src/uct/base/uct_cm.c

```diff
@@ -164,7 +164,8 @@ ucs_status_t uct_cm_check_ep_params(const uct_ep_params_t *params)
     return UCS_OK;
 }
 
-UCS_CLASS_INIT_FUNC(uct_cm_base_ep_t, const uct_ep_params_t *params)
+ucs_status_t uct_cm_use_ep_params(uct_cm_base_ep_t *ep,
```

Comment:
why not keep uct_cm_set_common_data() asis, and just add uct_cm_check_ep_params() to it's beginning?
also, seems this change is not really needed for supporting changing to another CM , it's just adding missing params check?

Reply:
@yosefe yes, it's done to avoid code duplication when calling these functions from `UCS_CLASS_INIT_FUNC(uct_cm_base_ep_t, const uct_ep_params_t *params)` and `uct_tcp_sockcm_ep_create`.
i'll move `uct_cm_check_ep_params` to the beginning of `uct_cm_set_common_data`